Server/JokesOnYou.Web.Api/Controllers/SavedJokeController.cs

```diff
@@ -11,5 +21,19 @@ public SavedJokeController(ISavedJokeService savedJokeService)
         {
             _savedJokeService = savedJokeService;
         }
+
+        [HttpPost("{jokeId}")]
+        public async Task<ActionResult> ToggleSavedJoke(int jokeId)
+        {
+            await _savedJokeService.ToggleSavedJoke(jokeId, ClaimsPrincipalExtension.GetUserId(User));
+            return NoContent();
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<JokeReplyDto>> GetSavedJokesByUserId()
+        {
+            var jokes = _savedJokeService.GetSavedJokesByUserId(ClaimsPrincipalExtension.GetUserId(User));
```

Comment:
Shouldn't this be awaited? I'm not sure how yield works?

Reply:
Yep, i've changed it